src/Plugin.php

```diff
@@ -13,47 +13,97 @@ class Plugin
+
+    /**
+     * @var Site
+     */
+    private $site;
+
+    /**
+     * @var SingleSwitcher
+     */
+    private $singleSwitcher;
+
+    /**
+     * @var Assets
+     */
+    private $assets;
+
+    /**
+     * @var Attachment
+     */
+    private $attachment;
+
+    /**
+     * @var Thumbnail
+     */
+    private $thumbnail;
+
     /**
      * Plugin constructor.
      *
      * @param $file
      */
     public function __construct(string $file)
     {
+        //init
         $this->rootFile = $file;
+        $this->pluginProperties = new PluginProperties($this->rootFile);
+        $this->site = new Site();
+        $this->singleSwitcher = new SingleSwitcher();
+        $this->assets = new Assets($this->pluginProperties);
+        $this->attachment = new Attachment($this->site, $this->singleSwitcher);
+        $this->thumbnail = new Thumbnail($this->site, $this->singleSwitcher);
     }
 
     /**
      * Integration the WordPress environment.
      */
     public function onLoad()
     {
-        $pluginProperties = new PluginProperties($this->rootFile);
-        $site = new Site();
-        $singleSwitcher = new SingleSwitcher();
-
-        $assets = new Assets($pluginProperties);
-        $attachment = new Attachment($site, $singleSwitcher);
-        $thumbnail = new Thumbnail($site, $singleSwitcher);
-
-        add_action('admin_enqueue_scripts', [$assets, 'enqueueScripts']);
-        add_action('admin_enqueue_scripts', [$assets, 'enqueueStyles']);
-        add_action('wp_ajax_query-attachments', [$attachment, 'ajaxQueryAttachments'], 0);
-        add_action('wp_ajax_get-attachment', [$attachment, 'ajaxGetAttachment'], 0);
-        add_action('wp_ajax_send-attachment-to-editor', [$attachment, 'ajaxSendAttachmentToEditor'], 0);
-        add_action('wp_get_attachment_image_src', [$attachment, 'attachmentImageSrc'], 99, 4);
-        add_filter('media_view_strings', [$attachment, 'mediaStrings']);
-
-        add_action('save_post', [$thumbnail, 'saveThumbnailMeta'], 99);
-        add_action('wp_ajax_get-post-thumbnail-html', [$thumbnail, 'ajaxGetPostThumbnailHtml'], 99);
-        add_filter('admin_post_thumbnail_html', [$thumbnail, 'adminPostThumbnailHtml'], 99, 3);
-        add_filter('post_thumbnail_html', [$thumbnail, 'postThumbnailHtml'], 99, 5);
-
-        if (\function_exists('wc')) {
-            $this->wcBootstrap($site, $singleSwitcher);
+
+
+        add_action('admin_enqueue_scripts', [$this->assets, 'enqueueScripts']);
+        add_action('admin_enqueue_scripts', [$this->assets, 'enqueueStyles']);
+        add_action('wp_ajax_query-attachments', [$this->attachment, 'ajaxQueryAttachments'], 0);
+        add_action('wp_ajax_get-attachment', [$this->attachment, 'ajaxGetAttachment'], 0);
+        add_action('wp_ajax_send-attachment-to-editor', [$this->attachment, 'ajaxSendAttachmentToEditor'], 0);
+        add_action('wp_get_attachment_image_src', [$this->attachment, 'attachmentImageSrc'], PHP_INT_MAX, 4);
```

Comment:
PHP_INT_MAX is not a solid option, especially for 3th level code there hook in our way. So please use her a integer, like 99. Thanks.